### ISSUE TITLE
`__slots__` is incorrectly classified as an instance variable across inheritance

### ISSUE THREAD

**Issue by XuehaiPan** [python/mypy#21893]
**Bug Report**

Mypy classifies `__slots__` as an instance variable even though Python defines it on the class. This causes a false-positive override error when a subclass explicitly annotates `__slots__` as a `ClassVar`.

The issue also affects plugin-generated `__slots__`, including slots generated by dataclasses and attrs.

**To Reproduce**

```python
from dataclasses import dataclass
from typing import ClassVar


@dataclass(slots=True)
class Base:
    x: float
    y: float


class Child(Base):
    __slots__: ClassVar[tuple[()]] = ()
```

Run:

```text
mypy example.py
```

Playground link: https://mypy-play.net/?mypy=latest&python=3.14&gist=e0e2f906ab9e0201c43d2b5cb6ff950c

**Expected Behavior**

Mypy should report no errors. Both generated and explicitly declared `__slots__` should be treated as class variables across inheritance.

**Actual Behavior**

```text
Cannot override instance variable (previously declared on base class "Base") with class variable  [misc]
```

A `# type: ignore[misc]` comment is currently required on the subclass declaration even though `__slots__` is a class-level attribute.

The same inconsistent classification occurs with ordinary `__slots__` declarations and attrs-generated slots.

**Your Environment**

- Mypy version used: 2.3.1
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.14.7

**Comment by bzoracler** [comment]
`__slots__` is only useful as effectively a `ClassVar[]`, that I agree with. But it is not a type error to treat it as an instance variable. You can arbitrarily assign `__slots__` at the instance level, no matter how weird it may look; the following works at runtime and mypy won't emit diagnostics:

```python
class A:
    __slots__ = ("__dict__",)

a = A()
a.__slots__ = ("hello",)
```

Like the implicitly-generated `__slots__` by `@dataclass(slots=True)`, the error `Cannot override instance variable with class variable` is the same as any other code which doesn't annotate `ClassVar[]` in a base class:

```python
from typing import ClassVar

class Base:
    a = 1

class Child(Base):
    a: ClassVar[int] = 2  # Cannot override instance variable (previously declared on base class "Base") with class variable  [misc]
```

If mypy emits diagnostics for `@dataclass(slots=True)` and incompatible child `ClassVar[]` annotations, it would be a linting rule rather than a type error. I have no real opinion on this as `__slots__` is special-cased to allow any `Iterable[str]` annotation in sub/superclass relations in an LSP-incompatible manner, so I've never added type annotations for `__slots__` anyway. Some people may want to annotate this if they're using the value of `ConcreteClass.__slots__` directly (dangerously LSP-incompatible though if `ConcreteClass: type[BaseClass]`), so I guess it may benefit them.
